Dockerfile

```diff
@@ -184,6 +185,15 @@ RUN --mount=target=/root/.cache,type=cache \
   xx-verify --static /out/containerd-stargz-grpc && \
   xx-verify --static /out/ctr-remote
 
+FROM gobuild-base AS nydus
+ARG NYDUS_VERSION
+ARG TARGETOS
+ARG TARGETARCH
+SHELL ["/bin/bash", "-c"]
+RUN set -e; [[ "$TARGETOS" == "linux" ]] && [[ "$TARGETARCH" == "amd64" || "$TARGETARCH" == "arm64" ]] || exit 1;
```

Comment:
This is likely to break the `integration-tests-base` on other platforms (arm, s390x, ppc64le, riscv64).

Can we just build nydus in this stage, or could you upload binaries for all the archs?

Reply:
> Added os/arch check, since nydus has only been tested in production on linux/amd64 and linux/arm64 machines, it is currently only guaranteed to be available on these two platforms, in the future we will add support for other linux archs or macOS, etc, thanks.

Need to depend on some rust components if build in this stage, as said here, how about letting us remove the arch check and keep available on amd64 and arm64 first? nydus will release more arches in the future.

Reply:
Will release more other platforms for making sure the integration tests are available as much as possible.